### ISSUE TITLE
data_element may be type tuple, instead of tf.Tensor

### ISSUE THREAD

**Issue by jtromans** [tensorflow/tensorflow#39377]
https://github.com/tensorflow/tensorflow/blob/2b96f3662bd776e277f86997659e61046b56c315/tensorflow/python/keras/engine/base_preprocessing_layer.py#L171

Is it expected that a `tf.data` Dataset should only contain features, without labels?

Concretely, `next_data()` would return a tuple if the Dataset contains separate features and labels (common when `model.fit(dataset, ...)` ). Due to the subsequent try-except, this will cause `shape` to be set to `None` (i.e. tuple has no attribute shape) and the preprocessing normalization layer will TypeError since `len(None)` etc.

Apologies if I've misunderstood.

**Comment by tanzhenyu** [comment]
It's expecting a dataset with a single input, it can be either feature or label
